### DISCUSSION TITLE
Webpack v5

### DISCUSSION THREAD

**Discussion by tujoworker** [gatsbyjs/gatsby#26146] [Ideas / Feature Requests] [answered]
## Summary

Webpack v5 (beta) is good on its way to get released.

So I think it's time to have a plan on how and when to integrate it in to Gatsby.
For me, a topic like that will naturally finds its place in the next major - v3.

### Motivation

There are a couple of nice features I think, also Gatsby would have benefits of:

- improve build performance with Persistent Caching.
- improve Long Term Caching with better algorithms and defaults.
- improve bundle size with better Tree Shaking and Code Generation

More details in the [Webpack v5 change log](https://github.com/webpack/changelog-v5)

Also, [Module Federation](https://github.com/module-federation) is something I see as a very interesting feature.

### More details

Next.js has just announced [their Webpack v5](https://nextjs.org/blog/next-9-5#webpack-5-support-beta) integration in that [blog post](https://nextjs.org/blog/next-9-5).

There is a [v4 to v5 migration guide](https://github.com/webpack/changelog-v5/blob/master/MIGRATION%20GUIDE.md).

**Comment by tujoworker** [comment, marked as answer]
Webpack 5 has been released 🥳🎉 – here's [the tweet](https://twitter.com/thelarkinn/status/1315062117221851136?s=12). Now – I really hope it will make it's way into Gatsby before v4 – because it will give definitively an improvement in bundle size from what all I read. Something which again will get better UX and Lighthouse results 💪

@wardpeet – is it listed now in the roadmap for the next major?

**Reply by rakthree** [reply]
Is there any update on this feature request? @tujoworker mentioned the Module Federation feature above, which will be very helpful and is what my team is looking to implement with Gatsby.

**Reply by tujoworker** [reply]
Gatsby v3 is out including Webpack v5 👏😍🎉

**Reply by steverandy** [reply]
Can someone explain how to use the persistent caching in webpack 5? I've tried adding the `cache` config to webpack config but it did not improve the build time.

**Reply by tujoworker** [reply]
Are you looking for incremental builds? https://www.gatsbyjs.com/docs/reference/release-notes/v3.0/#incremental-builds-in-oss

or the preserve Webpack cache feature?
https://www.gatsbyjs.com/docs/reference/release-notes/v2.28/#experimental-new-cache-clearing-behaviors

**Reply by steverandy** [reply]
Looking specifically for this: https://webpack.js.org/blog/2020-10-10-webpack-5-release/#persistent-caching

It seems that by default, it's not enabled in Gatsby v3. I've tried to override the cache config but not getting faster assets build. Curious if it can improve build time, because in Gatsby v3 I'm still getting build time over 10sec.
